osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/EventType.java

```diff
@@ -96,16 +96,33 @@
     CLOCK_ADJUSTED_M_BUS_CHANNEL_4(134),
     NEW_M_BUS_DEVICE_DISCOVERED_CHANNEL_4(135),
     PERMANENT_ERROR_FROM_M_BUS_DEVICE_CHANNEL_4(136),
-    MANUFACTURER_SPECIFIC_231(231);
+    MANUFACTURER_SPECIFIC_231(231),
+    MANUFACTURER_SPECIFIC_232(232),
+    MANUFACTURER_SPECIFIC_233(233),
+    MANUFACTURER_SPECIFIC_234(234),
+    MANUFACTURER_SPECIFIC_235(235),
+    MANUFACTURER_SPECIFIC_236(236),
+    MANUFACTURER_SPECIFIC_237(237),
+    MANUFACTURER_SPECIFIC_238(238),
+    MANUFACTURER_SPECIFIC_239(239),
+    MANUFACTURER_SPECIFIC_240(240),
+    MANUFACTURER_SPECIFIC_241(241),
+    MANUFACTURER_SPECIFIC_242(242),
+    MANUFACTURER_SPECIFIC_243(243),
+    MANUFACTURER_SPECIFIC_244(244),
+    MANUFACTURER_SPECIFIC_245(245),
+    MANUFACTURER_SPECIFIC_246(246),
+    MANUFACTURER_SPECIFIC_247(247),
+    MANUFACTURER_SPECIFIC_248(248),
+    MANUFACTURER_SPECIFIC_249(249);
 
-    private static final EventType[] VALUES = EventType.values();
-    private static final Map<Integer, EventType> lookup = new HashMap<>();
```

Comment:
IMO `lookup` is a better name than `eventCodes`. The Map does not contain eventCodes, but eventTypes and offers a lookup by their value. That said, I think a Map is not very useful here. Iterating over the values to find one is fine for an enum so the static block and the map could be removed and the getValue(int) method can be implemented with an iteration:

```
    for(EventType eventType : EventType.values()) {
        if (eventType.getValue() == intValue) {
            return eventType;
        }
    }
    throw new IllegalArgumentException("No EventType found for intValue " + intValue);
```

Note that the exception above is new; the current implementation simply returns null if the EventType is not found. If feasible the calling code should be refactored to handle such an exception. Another option is to return an Optional<EventType> instead of throwing an exception:

```
    public static Optional<EventType> getValue(final int intValue) {
        for(EventType eventType : EventType.values()) {
            if (eventType.getValue() == intValue) {
                return Optional.of(eventType);
            }
        }
       return Optional.empty();
    }
```

Returning null is a code smell.

Reply:
you are right, it contains eventTypes and not eventCodes. I mistakenly used codes because that was in the JIRA name.